cartographer/mapping_3d/pose_graph/constraint_builder.cc

```diff
@@ -244,8 +244,8 @@ void ConstraintBuilder::ComputeConstraint(
     if (match_full_submap) {
       info << " matches";
     } else {
-      const transform::Rigid3d difference = global_submap_pose.inverse() *
-                                            global_node_pose *
+      const transform::Rigid3d difference = global_node_pose.inverse() *
+                                            global_submap_pose *
                                             constraint_transform;
```

Comment:
drive-by comment: the comment that explains constraint_transform is ~50 lines above its declaration